JavaFXUserInterface/src/main/java/Controllers/SplashScreenController.java

```diff
@@ -0,0 +1,69 @@
+/*
+  This Controller is responsible for displaying the Splash Screen before displaying
+  before displaying the assets page.
+
+  @author Najim
+  @last_edit 11/03/2020
+ */
+package Controllers;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ResourceBundle;
+
+public class SplashScreenController implements Initializable {
+    @FXML
+    private VBox rootContainer;
+
+
+    /**
+     * Initialize runs before the scene is displayed.
+     * It initializes elements and data in the scene.
+     *
+     * @param url            url to be used
+     * @param resourceBundle resourceBundle to be used
+     * @author Najim
+     */
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        new SplashScreen().start();
+    }
+
+    class SplashScreen extends Thread {
+        @Override
+        public void run() {
+            try {
+                Thread.sleep(5500); //Displays the SplashScreen for 5.5 seconds
```

Comment:
I dont think there should be a hardcoded amount of time the splash screen is displayed, it should be displayed for as long as the app takes to boot up, no longer, no shorter

Reply:
Splash Screen is now implemented using a preloader with a fade transition. The Splash Screen is displayed as long as all of the assets are not loaded completely.